controllers/bookController.js

```diff
@@ -86,84 +88,83 @@ exports.book_create_get = function(req, res, next) {
 };
 
 // Handle book create on POST
-exports.book_create_post = function(req, res, next) {
-
-    req.checkBody('title', 'Title must not be empty.').notEmpty();
-    req.checkBody('author', 'Author must not be empty').notEmpty();
-    req.checkBody('summary', 'Summary must not be empty').notEmpty();
-    req.checkBody('isbn', 'ISBN must not be empty').notEmpty();
-
-    req.sanitize('title').escape();
-    req.sanitize('author').escape();
-    req.sanitize('summary').escape();
-    req.sanitize('isbn').escape();
-    req.sanitize('title').trim();
-    req.sanitize('author').trim();
-    req.sanitize('summary').trim();
-    req.sanitize('isbn').trim();
-     // Sanitize genre array for each value individually as validator works for string value only
-     if(req.body.genre instanceof Array){
-        req.body.genre = req.body.genre.map((initialGenre)=>{
-            req.body.tempGenre = initialGenre;
-            req.sanitize('tempGenre').escape();
-            return req.body.tempGenre;
-        });
-        delete req.body.tempGenre;
-        }else
-        req.sanitize('genre').escape();
-
-    var book = new Book(
-      { title: req.body.title,
-        author: req.body.author,
-        summary: req.body.summary,
-        isbn: req.body.isbn,
-        genre: (typeof req.body.genre==='undefined') ? [] : req.body.genre
-       });
-
-    console.log('BOOK: '+book);
-
-    var errors = req.validationErrors();
-    if (errors) {
-        // Some problems so we need to re-render our book
-        console.log('GENRE: '+req.body.genre);
-
-        console.log('ERRORS: '+errors);
-        //Get all authors and genres for form
-        async.parallel({
-            authors: function(callback) {
-                Author.find(callback);
-            },
-            genres: function(callback) {
-                Genre.find(callback);
-            },
-        }, function(err, results) {
-            if (err) { return next(err); }
+exports.book_create_post = [
+   
+    // Validate fields
+    body('title', 'Title must not be empty.').isLength({ min: 1 }).trim(),
+    body('author', 'Author must not be empty.').isLength({ min: 1 }).trim(),
+    body('summary', 'Summary must not be empty.').isLength({ min: 1 }).trim(),
+    body('isbn', 'ISBN must not be empty').isLength({ min: 1 }).trim(),
+
+    // Sanitize fields
+    sanitizeBody('title').trim().escape(),
+    sanitizeBody('author').trim().escape(),
+    sanitizeBody('summary').trim().escape(),
+    sanitizeBody('isbn').trim().escape(),
+    
+    // Process request after validation and sanitization
+    (req, res, next) => {
+        
+        // Sanitize genre array for each value individually (validator only works for strings)
```

Comment:
@VaibhavAjayGupta I used the legacy API here. I would appreciate your advice on the best way to do this on the new API!

Reply:
My suggestion is #31